base/mem_trie.hpp

```diff
@@ -170,8 +224,7 @@ class MemTrie
   template <typename ToDo>
   void ForEachInNode(Node const & node, String const & prefix, ToDo && toDo) const
   {
-    for (auto const & value : node.m_values)
-      toDo(prefix, value);
+    node.m_values.ForEach(std::bind(std::forward<ToDo>(toDo), prefix, std::placeholders::_1));
```

Comment:
Errr, unfortunately we don't use cpp14, that's why I can't use auto in lambdas. But in the case of bind, thanks, it's a right thing to use ref here.